mscore/instrwidget.cpp

```diff
@@ -897,6 +911,12 @@ void InstrumentsWidget::on_search_textChanged(const QString &searchPhrase)
 
 void InstrumentsWidget::on_instrumentGenreFilter_currentIndexChanged(int index)
       {
+      QSettings settings;
+      if (storeChangeInGenre){
+            settings.setValue("selectedInstrumentGenre", index);
+            settings.sync();
+            }
```

Comment:
couldn't these settings be saved in the destructor?
They would only be saved once, that way.

Reply:
That's a good point actually. I can't test it right now, but I seem to remember that if you access the value of a setting after it's been changed but not synced, it will have the last synced value. The variable would be unneeded, I guess because:

1. The widget is initialized 
2. The genre settings value is changed a couple of times automatically but not synced
3. The genre filter is set to the genre settings value (but the last saved one, which would be user chosen)
4. This change in genre filter is saved to the genre settings variable (meaning that any automatic genre changes are basically ignored) 
5. User possibly changes genre
6. Widget closes, last chosen genre is saved

Sounds good to me. Basically, you're right :)